### DISCUSSION TITLE
Please consider something that runs basic valid `perl` scripts?

### DISCUSSION THREAD

**Discussion by oodler577** [uutils/coreutils#6673] [Ideas]
See: https://www.reddit.com/r/perl/comments/1f2s80z/comment/lkat8ot/

I am a Perl programmer, not a Rust programmer. But I see value is something that can run a strict subset of Perl for simple baby Perl scripts. 

As long as anything a Rust inplementation of `perl` is valid in the One True^tm `perl`, then I think this would bring value to a lot of people as a "drop-in". It doesn't even have to do fancy things like `use` modules. Keep it simple and strictly adherent to what might be valid to what actual `perl` can run, and it will be fine.

It will generate a lot of enthusiam in the Perl community for Rust potentially and also invigorate interest in Perl - as simpe as the scripts this might support would be. Thank you.

**Comment by tertsdiepraam** [comment]
That sounds pretty complicated actually. Interpreters tend to get complex pretty quickly. I think this would be better suited for a separate project. Also, I personally have no familiarity with Perl and have never used it, so I don't think I would be a good fit to write something like this. The other maintainers might feel differently though.

If I were you, I'd keep asking around to find people who are active in both the Rust and Perl communities.

**Comment by cakebaker** [comment]
I agree with @tertsdiepraam and I think it's outside the scope of this project.
